sorc/global_cycle.fd/land_increments.f90

```diff
@@ -549,30 +730,40 @@ subroutine apply_land_da_adjustments_soil(lsm, isot, ivegsrc,lensfc, &
 
         if (upd_stc) then
 
-          print *, 'Reverting frozen noah-mp model stc back to background'
-          n_revert=0
+          call set_soilveg_noahmp(isot, ivegsrc, maxsmc, bb, satpsi, iret)
+          if (iret < 0) then
+               print *, 'FATAL ERROR: problem in set_soilveg_noahmp'
+               call mpi_abort(mpi_comm_world, 10, ierr)
+          endif
+
+          print *, 'Adjusting noah-mp soil moisture to be consistent with soil temperature update' 
           n_stc = 0
           n_slc = 0
 
           do i=1,lensfc
-          if (stc_updated(i) == 1 ) then
+            if (stc_updated(i) == 1 ) then ! soil-only location
```

Comment:
I haven't gone through the logic here.

Reply:
Hi Mike @barlage, Clara mentioned that she would like you to review this part of the code (i.e., consistency check). Thanks!!